postprocess_segmentation.py

```diff
@@ -498,12 +524,77 @@ def main(params):
                 logging.error(f"\nError generalizing using {cont_type} container with image {gen_cont_image}."
                               f"\ncommand: {command}"
                               f"\nError {type(e)}: {e}")
+            # Collect the name of the layer created
+            if 'ROAI' in classes_dict: # take the polygon layer for the road
+                m = re.search('outlayernamepoly=(.{6})', str(command))
+            else:
+                m = re.search('outlayername=(.{6})', str(command))
+            layer_name = m.group(1)
+            
         if out_gen.is_file():
             logging.info(f'\nGeneralization completed. Final prediction: {out_gen}')
             returned_vector_pred = out_gen
+            
+            try: # Try to convert multipolygon to polygon
+                df = geopandas.read_file(returned_vector_pred, layer=layer_name)
+                if 'MultiPolygon' in df['geometry'].geom_type.values:
+                    logging.info("\nConverting multiPolygon to Polygon...")
+                    gdf_exploded = df.explode(index_parts=True, ignore_index=True)
+                    gdf_exploded.to_file(returned_vector_pred, layer=layer_name)
+            except Exception as e:
+                logging.error(f"\nSomething went wrong during the convertion of Polygon. \nError {type(e)}: {e}")
```

Comment:
Could this become a function of its own?

Reply:
done